src/wasm2c.c.tmpl

```diff
@@ -22,8 +22,8 @@
        || TRAP(CALL_INDIRECT)                        \
        , ((t)table.data[x].func)(__VA_ARGS__))
 
-#define RANGE_CHECK(mem, a, t) \
-  if (UNLIKELY((a) + sizeof(t) > mem->size)) TRAP(OOB)
+#define RANGE_CHECK(mem, addr, len) \
```

Comment:
The MEMCHECK macro (line 31) uses RANGE_CHECK with a typename when doing software bounds-checks. L31 could be updated to something like `#define MEMCHECK(mem, a, t) RANGE_CHECK(mem, a, sizeof(t))` to keep this working on the non-signal-handler platforms. (Also I think this needs a `make update-wasm2c` for these changes to be synced to the prebuilt headers and tested.)